subgraphs/opyn-gamma/src/entities/event.ts

```diff
@@ -0,0 +1,90 @@
+import { Address, BigInt, Bytes, ethereum } from "@graphprotocol/graph-ts";
+import { NetworkConfigs } from "../../configurations/configure";
+import { Deposit, Withdraw } from "../../generated/schema";
+import { getOrCreateToken } from "../common/tokens";
+import { getUSDAmount } from "../price";
+import { prefixID } from "../common/utils/strings";
+import { getOrCreateAccount } from "./account";
+import { handlePoolDeposit, getOrCreatePool, handlePoolWithdraw } from "./pool";
+import { getOrCreateOpynProtocol } from "./protocol";
+import {
+  incrementProtocolDepositCount,
+  incrementProtocolWithdrawCount,
+} from "./usage";
+import { takeSnapshots } from "./snapshot";
+
+export function createDeposit(
+  event: ethereum.Event,
+  asset: Address,
+  amount: BigInt,
+  from: Address
+): void {
+  const account = getOrCreateAccount(from);
+  const token = getOrCreateToken(asset);
+  const pool = getOrCreatePool(token);
+  takeSnapshots(event, pool);
```

Comment:
This is how snapshots are done in the sdk. It's not perfect and the daily fields are not completely accurate. But doing it after would still miss any deposits made on the same day after the snapshot is taken, since the snapshots are immutable now.

Reply:
Ah I think I interpreted this wrong. If I understand correctly now, this will actually be accurate bc the snapshot is taken beforehand IF the current event is on a new day, so it will update a previous day with all events that occurred before the event being handled.

Reply:
You're right in that a snapshot is taken whenever 1 or 24 hours have passed since the last one, but previous (already taken) ones can't be updated, since they're immutable. All the events that happened after the last snapshot will be included in the next one, whenever that is taken.